zuul-core/src/main/java/com/netflix/zuul/message/http/HttpQueryParams.java

```diff
@@ -176,30 +187,34 @@ public void clear()
 
     public String toEncodedString()
     {
-        StringBuilder sb = new StringBuilder();
-        try {
-            for (Map.Entry<String, String> entry : entries()) {
-                sb.append(URLEncoder.encode(entry.getKey(), "UTF-8"));
-                if (StringUtils.isNotEmpty(entry.getValue())) {
-                    sb.append('=');
-                    sb.append(URLEncoder.encode(entry.getValue(), "UTF-8"));
+        if (propertyFactory.getBooleanProperty(ZuulConstants.ZUUL_KEEP_ORIGINAL_QUERY_STRING_ENCODING,false).getValue()) {
```

Comment:
nit: this should not be dynamically changeable.   You should capture the value when the query string is parsed originally, and use that to decide what the string value should be.